eth/handler.go

```diff
@@ -831,17 +831,32 @@ type NodeInfo struct {
 	Genesis    common.Hash         `json:"genesis"`    // SHA3 hash of the host's genesis block
 	Config     *params.ChainConfig `json:"config"`     // Chain configuration for the fork rules
 	Head       common.Hash         `json:"head"`       // SHA3 hash of the host's best owned block
+	Consensus  string              `json:"consensus"`  // Consensus mechanism in use
 }
 
 // NodeInfo retrieves some protocol metadata about the running host node.
 func (pm *ProtocolManager) NodeInfo() *NodeInfo {
 	currentBlock := pm.blockchain.CurrentBlock()
+
+	// Set up the Consensus field
+	consensus := "unknown"
+	if pm.blockchain.Config().Clique != nil {
+		consensus = "clique"
+	} else if pm.blockchain.Config().Istanbul != nil {
+		consensus = "istanbul"
+	} else if pm.blockchain.Config().Ethash != nil {
+		consensus = "ethash"
+	} else if pm.raftMode {
+		consensus = "raft"
+	}
+
```

Comment:
i think it would be more robust to assert on the engine type rather than config
```go
	if pm.raftMode { // raft does not using consensus interface
		consensusAlgo = "raft"
	} else {
		switch pm.engine.(type) {
		case consensus.Istanbul:
			consensusAlgo = "istanbul"
		case *clique.Clique:
			consensusAlgo = "clique"
		case *ethash.Ethash:
			consensusAlgo = "ethash"
		default:
			consensusAlgo = "unknown"
		}
	}
```

And if this could be extracted to a separate method, we can easily write unit tests for it